hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/FileIndex.java

```diff
@@ -309,6 +287,13 @@ private DataPruner initializeDataPruner(List<ResolvedExpression> filters) {
       validateConfig();
       return null;
     }
-    return DataPruner.newInstance(filters);
+    return dataPruner;
+  }
+
+  private void validateConfig() {
```

Comment:
`validateConfig` -> `isDataSkippingFeasible`, and returns boolean instead of void.